packages-exp/auth-types-exp/index.d.ts

```diff
@@ -327,152 +667,439 @@ export abstract class EmailAuthProvider implements AuthProvider {
+ * // User signed in.
+ * ```
+ *
+ * @public
  */
 export abstract class MultiFactorResolver {
+  /**
+   * The list of hints for the second factors needed to complete the sign-in for the current
+   * session.
+   */
   hints: MultiFactorInfo[];
+  /**
+   * The session identifier for the current sign-in flow, which can be used to complete the second
+   * factor sign-in.
+   */
   session: MultiFactorSession;
+  /**
+   * A helper function to help users complete sign in with a second factor using an
+   * {@link MultiFactorAssertion} confirming the user successfully completed the second factor
+   * challenge.
+   *
+   * @param assertion - The multi-factor assertion to resolve sign-in with.
+   * @returns The promise that resolves with the user credential object.
+   */
   resolveSignIn(assertion: MultiFactorAssertion): Promise<UserCredential>;
 }
 
 /**
- * https://firebase.google.com/docs/reference/js/firebase.auth.multifactorsession
+ * The multi-factor session object used for enrolling a second factor on a user or helping sign in
+ * an enrolled user with a second factor.
+ *
+ * @public
  */
 export interface MultiFactorSession {}
 
 /**
- * https://firebase.google.com/docs/reference/js/firebase.user.multifactoruser
+ * This is the interface that defines the multi-factor related properties and operations pertaining
+ * to a {@link User}.
+ *
+ * @public
  */
 export interface MultiFactorUser {
+  /** Returns a list of the user's enrolled second factors. */
   readonly enrolledFactors: MultiFactorInfo[];
+  /**
+   * Returns the session identifier for a second factor enrollment operation. This is used to
+   * identify the user trying to enroll a second factor.
+   *
+   * @returns The promise that resolves with the {@link MultiFactorSession}.
+   */
   getSession(): Promise<MultiFactorSession>;
+  /**
+   *
+   * Enrolls a second factor as identified by the {@link MultiFactorAssertion} for the
+   * user. On resolution, the user tokens are updated to reflect the change in the JWT payload.
+   * Accepts an additional display name parameter used to identify the second factor to the end
+   * user. Recent re-authentication is required for this operation to succeed. On successful
+   * enrollment, existing Firebase sessions (refresh tokens) are revoked. When a new factor is
+   * enrolled, an email notification is sent to the user’s email.
+   *
+   * @example
+   * ```
+   * const multiFactorUser = multiFactor(auth.currentUser);
+   * const multiFactorSession = await multiFactorUser.getSession();
+   * // Send verification code
+   * const phoneAuthProvider = new PhoneAuthProvider();
+   * const phoneInfoOptions = {
+   *   phoneNumber: phoneNumber,
+   *   session: multiFactorSession
+   * };
+   * const verificationId = await phoneAuthProvider.verifyPhoneNumber(phoneInfoOptions, appVerifier);
+   * // Store verificationID and show UI to let user enter verification code.
```

Comment:
Suggest backticks if this is a literal.

Just "verification ID" in English works fine for me fwiw.